src/scheduler/worker.rs

```diff
@@ -3,30 +3,42 @@ use super::work_bucket::*;
 use super::*;
 use crate::mmtk::MMTK;
 use crate::util::opaque_pointer::*;
+use crate::vm::{Collection, VMBinding};
 use std::ffi::c_void;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::mpsc::Sender;
 use std::sync::{Arc, Weak};
 
+/// Thread-local data for each worker thread.
+///
+/// For mmtk, each gc can define their own worker-local data, to contain their required copy allocators and other stuffs.
+pub trait GCWorkerLocal {
+    fn init(&mut self, _tls: VMWorkerThread) {}
+}
+
+/// A default implementation for scheduling systems that does not require a worker-local context.
+impl GCWorkerLocal for () {}
```

Comment:
Do we still need this?

Reply:
Thanks for pointing this out. We don't need this anymore. I've removed it.